src/storage/heap_file.c

```diff
@@ -25255,10 +25255,18 @@ heap_get_visible_version_internal (THREAD_ENTRY * thread_p, HEAP_GET_CONTEXT * c
       snapshot_res = mvcc_snapshot->snapshot_fnc (thread_p, &mvcc_header, mvcc_snapshot);
       if (snapshot_res == TOO_NEW_FOR_SNAPSHOT)
 	{
+	  const LOG_LSA *prev_lsa = &MVCC_GET_PREV_VERSION_LSA (&mvcc_header);
+	  if (prev_lsa->is_null ())
```

Comment:
But, I think we cannot because the REC_BIGONE type. So, how about remove the assert() and add the if (prev_lsa->is_null ()) condition in the heap_get_visible_version_from_log() function.

Reply:
@hornetmj @wrlawodms 
Please be aware that there is also a related issue which is a regression coming from QA - http://jira.cubrid.org/browse/LETS-532 - and the test bumped into the assert. So the assert must be evaluated also in the context of this regression (and, possibly others) in the most conservative way.
Me, personally, I would leave the assert if it does not offend any other tests.

Reply:
If the `previous_version_lsa` is null inside the `heap_get_visible_version_from_log()` function, it does nothing(Line 25000) and throws an error (S_DOESNT_EXIST). Rather than adding asserts in `heap_get_visible_version_from_log()`, how about adding exception handling earlier?

Reply:
@hornetmj 
>In develop branch, the heap_get_visible_version_from_log() can handle when the prev_lsa is NULL.

I don't think it was handled correctly because it used the NULL previous LSA to compare with:
```if (LSA_LT (&oldest_prior_lsa, previous_version_lsa))```
, which looks unreasonable and causes unnecessary appending prior lists.

>But, I think we cannot because the REC_BIGONE type.

The big one record can have the NULL previous LSA, and other types also can be considered to have the NULL lsa here (the flag is off). However, it's only the case when the record is newly inserted. In this case, it seems unreasonable to find the previous version. On top of that, the semantics at all seems awkward to find a previous version in the LOG with NULL LSA (N/A location);

Reply:
@joohok You mean the exception handler inside the function like hornetmj's opinion?

Reply:
@hornetmj oh sorry to confuse you. It doesn't trigger.
> which looks unreasonable and causes unnecessary appending prior lists.

Reply:
> On top of that, the semantics at all seems awkward to find a previous version in the LOG with NULL LSA (N/A location);

I think this is right. Now I agree with your change. I think exception handler is unnecessary.

Reply:
@wrlawodms @joohok I understand what you guies mean. However, in my point of view, it is better to handle in the heap_get_visible_version_from_log() from the heap_* API user's point. From a semantic point of view, I think there is no problem. However, I will follow the comments of the two reviewers.